### ISSUE TITLE
S3 repository not respecting endpoint param

### ISSUE THREAD

**Issue by bputt** [crate/crate#7595]
**CrateDB version**: 3.0.5

**Environment description**:

Centos 7.latest
Openjdk 1.8.0_181

**Problem description**:
When setting up a repository, specifying the endpoint is not being used and the default is only taken into account

**Steps to reproduce**:

```
CREATE REPOSITORY mybackup TYPE s3 WITH (bucket='my-backup', endpoint='https://some-endpoint/', protocol='https')
```

Relevant log message on master:

```
failed to create repository [s3][mybackup]
com.amazonaws.SdkClientException: Unable to execute HTTP request: mybackup.s3.amazonaws.com
at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1114) ~[aws-java-sdk-core-1.11.223.jar:?]
```

As you can see, it's still trying to use ```s3.amazonaws.com``` instead of ```https://some-endpoint/```

**Comment by izloop** [comment]
Hi, could it be that you have made an error when writing the `CREATE REPOSITORY` command?

Please check this documentation and see if your problem persists, 
https://crate.io/docs/crate/reference/en/latest/sql/statements/create-repository.html#s3
Thanks

**Comment by bputt** [comment]
It's possible that it could be user error, however, the same command worked on an earlier version of Crate 1.0.2. I also provided the statement that I'm using and it seems inline with the documentation you posted

**Comment by izloop** [comment]
Could you please try this for me

`CREATE REPOSITORY mybackup TYPE s3 [ WITH (bucket='my-backup', endpoint='https://some-endpoint/', protocol='https') ]`

**Comment by izloop** [comment]
Hi @bputt - I have reviewed the issue and since I am able to reproduce it as well, this shall be escalated! I will feedback you any update we have on this here! 

Thanks for understanding.

**Comment by seut** [comment]
@bputt We've fixed it by https://github.com/crate/crate/pull/7635, it will be released with the next 3.0 and 3.1 hotfix releases. Thanks a lot for reporting.
